### PR TITLE
consume topic

### PR DESCRIPTION
**Pull request by brunoroger**

### CODE REVIEW

File: src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt [hunk 1]
```diff
@@ -5,31 +5,29 @@ import br.com.zup.op.events.domain.EventRepository
...
-import kotlin.collections.ArrayList
+
 
 @Service
 class KafkaEventManager(
         private val eventRepository: EventRepository,
         private val reasonRepository: ReasonRepository,
-        private val topicRepository: TopicRepository,
-        private val kafkaProducerConfig: KafkaProducerConfig) : EventManager {
+        private val kafkaProducerConfig: KafkaProducerConfig,
```
Comment (sandokandias) [inline src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt]: Remove the paramenter "kafkaProducerConfig".

File: src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt [hunk 2]
```diff
@@ -54,7 +52,17 @@ class KafkaEventManager(
     }
 
     override fun topicList(): ArrayList<TopicEntiy> {
-        return topicRepository.findAll() as ArrayList<TopicEntiy>
+        var listTopics  = topicConsumer.listTopics()
+
```
Comment (sandokandias) [inline src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt]: Break lines is not necessary in small contexts.

File: src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt [hunk 3]
```diff
@@ -54,7 +52,17 @@ class KafkaEventManager(
     }
 
     override fun topicList(): ArrayList<TopicEntiy> {
```
Comment (sandokandias) [inline src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt]: the name of function could be `listTopics`.

File: src/main/kotlin/br/com/zup/op/events/producer/KafkaProducerConfig.kt [hunk 4]
```diff
@@ -12,6 +15,8 @@ import java.util.*
 @Configuration
 class KafkaProducerConfig {
```
Comment (sandokandias) [inline src/main/kotlin/br/com/zup/op/events/producer/KafkaProducerConfig.kt]: Rename `KafkaProducerConfig` to `KafkaConfig`

File: src/main/kotlin/br/com/zup/op/events/producer/KafkaProducerConfig.kt [hunk 5]
```diff
@@ -12,6 +15,8 @@ import java.util.*
...
 
+  @Autowired
+  private val consumerFactory: ConsumerFactory<String, Any>? = null
 
   //@Value("${spring.kafka.bootstrapAddress}")
```
Comment (sandokandias) [inline src/main/kotlin/br/com/zup/op/events/producer/KafkaProducerConfig.kt]: Remove the comment, the `val`, the `KafkaTemplate`

**Review by sandokandias — CHANGES_REQUESTED** [review]

File: src/main/kotlin/br/com/zup/op/events/producer/KafkaProducerConfig.kt (hunk 5 above)
Comment (brunoroger) [reply src/main/kotlin/br/com/zup/op/events/producer/KafkaProducerConfig.kt]: The attribute will be reviewed after @Willcd  will commit a code publishing event

File: src/main/kotlin/br/com/zup/op/events/producer/KafkaProducerConfig.kt (hunk 4 above)
Comment (brunoroger) [reply src/main/kotlin/br/com/zup/op/events/producer/KafkaProducerConfig.kt]: The class will be rename after @Willcd  commit a code publish event

File: src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt (hunk 1 above)
Comment (brunoroger) [reply src/main/kotlin/br/com/zup/op/events/application/KafkaEventManager.kt]: The parameter will be removed after @Willcd  commit a code publish event

**Pull request merged**